File: tests/verify_mobile.py

import asyncio
from playwright.async_api import async_playwright

async def run():
    async with async_playwright() as p:
        iphone_12 = p.devices['iPhone 12']
        browser = await p.chromium.launch()
        context = await browser.new_context(
            **iphone_12
        )
        page = await context.new_page()

        print('Navigating to models.html...')
        await page.goto('http://localhost:3000/models.html?mobile=true')

        # Wait for the hub to appear
        print('Waiting for mobile hub...')
        await page.wait_for_selector('.gh-mobile-overlay', timeout=10000)

        # Capture English screenshot
        print('Capturing English screenshot...')
        await page.screenshot(path='hub_en.png')

        # Toggle to Spanish
        print('Toggling language...')
        await page.click('#gh-mobile-lang-btn')

        # Give it a moment to update
        await page.wait_for_timeout(500)

        # Capture Spanish screenshot
        print('Capturing Spanish screenshot...')
        await page.screenshot(path='hub_es.png')

        # Check for specific text to verify
        hub_title = await page.inner_text('.gh-hub-title')
        print(f'Hub Title (ES): {hub_title}')

        lang_btn_text = await page.inner_text('#gh-mobile-lang-btn')
        print(f'Lang Button Text (ES): {lang_btn_text}')

        model_title = await page.inner_text('.gh-model-title')
        print(f'First Model Title (ES): {model_title}')

        select_btn_text = await page.inner_text('.gh-mobile-btn')
        print(f'Select Button Text (ES): {select_btn_text}')

        await browser.close()

if __name__ == '__main__':
    asyncio.run(run())

File: tests/verify_mobile.js

const playwright = require('playwright');

(async () => {
    const browser = await playwright.chromium.launch();
    const context = await browser.newContext({
        ...playwright.devices['iPhone 12'],
        viewport: { width: 390, height: 844 }
    });
    const page = await context.newPage();

    console.log('Navigating to models.html...');
    await page.goto('http://localhost:3000/models.html?mobile=true');

    // Wait for the hub to appear
    console.log('Waiting for mobile hub...');
    await page.waitForSelector('.gh-mobile-overlay', { timeout: 10000 });

    // Capture English screenshot
    console.log('Capturing English screenshot...');
    await page.screenshot({ path: 'hub_en.png' });

    // Toggle to Spanish
    console.log('Toggling language...');
    await page.click('#gh-mobile-lang-btn');

    // Give it a moment to update
    await page.waitForTimeout(500);

    // Capture Spanish screenshot
    console.log('Capturing Spanish screenshot...');
    await page.screenshot({ path: 'hub_es.png' });

    // Check for specific text to verify
    const hubTitle = await page.innerText('.gh-hub-title');
    console.log('Hub Title (ES):', hubTitle);

    const langBtnText = await page.innerText('#gh-mobile-lang-btn');
    console.log('Lang Button Text (ES):', langBtnText);

    const modelTitle = await page.innerText('.gh-model-title');
    console.log('First Model Title (ES):', modelTitle);

    const selectBtnText = await page.innerText('.gh-mobile-btn');
    console.log('Select Button Text (ES):', selectBtnText);

    await browser.close();
})();
